File: aplusb2.java
import java.util.*;
import java.io.*;
public class apb {
	public static void main(String[] args) throws Exception {
		/*
		System.out.println(isSmaller("11","5")); // -1
		System.out.println(isSmaller("5","11")); // 1
		System.out.println(isSmaller("5","6")); // 1
		System.out.println(isSmaller("6","5")); // -1
		System.out.println(isSmaller("5","5")); // 0
		*/
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		for (int i = 0; i < n; i++) {
			String[] line = br.readLine().split(" ");
			System.out.println(APlusB(line[0], line[1]));
		}
	}
	
	public static String APlusB(String a, String b) {
		String res = "";
		boolean aIsNeg = a.charAt(0) == '-';
		boolean bIsNeg = b.charAt(0) == '-';
		if (aIsNeg == bIsNeg) {
			if (aIsNeg) {
				res = "-" + trimZeros(add(a.substring(1), b.substring(1)));
			}
			else {
				res = trimZeros(add(a, b));
			}
		}
		else {
			if (aIsNeg) {
				a = a.substring(1);
				int smaller = isSmaller(b, a);
				if (smaller > 0) {
					res = "-" + trimZeros(sub(a, b));
				}
				else if (smaller < 0) {
					res = trimZeros(sub(b, a));
				}
				else {
					return "0";
				}
			}
			else {
				b = b.substring(1);
				int smaller = isSmaller(b, a);
				if (smaller > 0) {
					res = trimZeros(sub(a, b));
				}
				else if (smaller < 0) {
					res = "-" + trimZeros(sub(b, a));
				}
				else {
					return "0";
				}
			}
		}
		
		return res;
	}
	
	public static int isSmaller(String a, String b) {
		int lenA = a.length();
		int lenB = b.length();
		
		if (lenA < lenB) {
			return 1;
		}
		
		if (lenA > lenB) {
			return -1;
		}
		
		for (int i = 0; i < lenA; i++) {
			if (a.charAt(i) < b.charAt(i)) {
				return 1;
			}
			if (a.charAt(i) > b.charAt(i)) {
				return -1;
			}
		}
		
		return 0;
	}
	
	public static String trimZeros(String str) {
		char[] chars = str.toCharArray();
		int s;
		for (s = 0; s < chars.length; s++) {
			if (chars[s] != '0') {
				break;
			}
		}
		return new String(Arrays.copyOfRange(chars, s, chars.length));
	}
	
	public static String add(String a, String b) {
		String res = "";
		int lenA = a.length();
		int lenB = b.length();
		if (lenA > lenB) {
			String t = a;
			a = b;
			b = t;
			int nt = lenA;
			lenA = lenB;
			lenB = nt;
		}
		
		int lenDiff = lenB - lenA;
		int carry = 0;
		for (int i = lenA- 1; i >= 0; i--) {
			int sum = (a.charAt(i) - '0') + (b.charAt(i + lenDiff) - '0') + carry;
			res += (char)(sum % 10 + '0');
			carry = sum / 10;
		}
		
		for (int i = lenDiff-1; i >= 0; i--) {
			int sum = b.charAt(i) - '0' + carry;
			res += (char)(sum % 10 + '0');
			carry = sum / 10;
		}
		
		if (carry > 0) {
			res += (char)(carry + '0');
		}
		
		return new StringBuilder(res).reverse().toString();
	}
	
	public static String sub(String a, String b) {
		String res = "";
		int lenA = a.length();
		int lenB = b.length();
		int lenDiff = lenA - lenB;
		int carry = 0;
		
		for (int i = lenB-1; i >= 0; i--) {
			int diff = (a.charAt(i + lenDiff) - '0') - (b.charAt(i) - '0') - carry;
			if (diff < 0) {
				diff += 10;
				carry = 1;
			}
			else {
				carry = 0;
			}
			res += String.valueOf(diff);
		}
		
		for (int i = lenDiff-1; i >= 0; i--) {
			if (a.charAt(i) == '0' && carry > 0) {
				res += "9";
				continue;
			}
			int diff = a.charAt(i) - '0' - carry;
			if (i > 0 || diff > 0) {
				res += String.valueOf(diff);
			}
			carry = 0;
		}
		return new StringBuilder(res).reverse().toString();
	}
}

File: aplusb2.cpp
#include <bits/stdc++.h>
using namespace std;


// itoa is non standard
// https://www.geeksforgeeks.org/implement-itoa/
char* itoa(int num, char* str, int base) {
    int i = 0;
    bool isNegative = false;
    if (num == 0) {
        str[i++] = '0';
        str[i] = '\0';
        return str;
    }
    if (num < 0 && base == 10) {
		isNegative = true;
		num = -num;
    }
    while (num != 0) {
		int rem = num % base;
		str[i++] = (rem > 9)? (rem-10) + 'a' : rem + '0';
		num = num/base;
    }
	if (isNegative)
		str[i++] = '-';
	str[i] = '\0';
	int start = 0; 
    int end = i - 1; 
    while (start < end) {
        swap(*(str+start), *(str+end));
        start++;
        end--;
    }
	return str;
}





string sub(string a, string b) {
	char buff[10];
	string res = "";
	int lenA = a.length();
	int lenB = b.length();
	int lenDiff = lenA - lenB;
	int carry = 0;
	
	for (int i = lenB-1; i >= 0; i--) {
		int diff = (a[i + lenDiff] - '0') - (b[i] - '0') - carry;
		if (diff < 0) {
			diff += 10;
			carry = 1;
		}
		else {
			carry = 0;
		}
		itoa(diff, buff, 10);
		res += buff;
	}
	
	for (int i = lenDiff-1; i >= 0; i--) {
		if (a[i] == '0' && carry > 0) {
			res += "9";
			continue;
		}
		int diff = a[i] - '0' - carry;
		if (i > 0 || diff > 0) {
			itoa(diff, buff, 10);
			res += buff;
		}
		carry = 0;
	}
	return string(res.rbegin(), res.rend());
}

string add(string a, string b) {
	string res = "";
	int lenA = a.length();
	int lenB = b.length();
	if (lenA > lenB) {
		swap(a, b);
		swap(lenA, lenB);
	}
	
	int lenDiff = lenB - lenA;
	int carry = 0;
	for (int i = lenA- 1; i >= 0; i--) {
		int sum = (a[i] - '0') + (b[i + lenDiff] - '0') + carry;
		res += (char)(sum % 10 + '0');
		carry = sum / 10;
	}
	
	for (int i = lenDiff-1; i >= 0; i--) {
		int sum = b[i] - '0' + carry;
		res += (char)(sum % 10 + '0');
		carry = sum / 10;
	}
	
	if (carry > 0) {
		res += (char)(carry + '0');
	}
	
	return string(res.rbegin(), res.rend());
}

string trimZeros(string str) {
	int i = 0;
	while (i < str.size() && str[i] == '0')
		++i;
	return str.substr(i);
}

int isSmaller(string a, string b) {
	int lenA = a.length();
	int lenB = b.length();
	
	if (lenA < lenB)
		return 1;
	
	if (lenA > lenB)
		return -1;
	
	for (int i = 0; i < lenA; i++) {
		if (a[i] < b[i])
			return 1;
		if (a[i] > b[i])
			return -1;
	}
	
	return 0;
}

string apb(string a, string b) {
	string res = "";
	bool aIsNeg = a[0] == '-';
	bool bIsNeg = b[0] == '-';
	if (aIsNeg == bIsNeg)
		if (aIsNeg)
			res = "-" + trimZeros(add(a.substr(1), b.substr(1)));
		else
			res = trimZeros(add(a, b));
	
	else
		if (aIsNeg) {
			a = a.substr(1);
			int smaller = isSmaller(b, a);
			if (smaller > 0) {
				res = "-" + trimZeros(sub(a, b));
			}
			else if (smaller < 0) {
				res = trimZeros(sub(b, a));
			}
			else {
				return "0";
			}
		}
		else {
			b = b.substr(1);
			int smaller = isSmaller(b, a);
			if (smaller > 0) {
				res = trimZeros(sub(a, b));
			}
			else if (smaller < 0) {
				res = "-" + trimZeros(sub(b, a));
			}
			else {
				return "0";
			}
		}
	
	return res;
}

int main() {
	int n;
	cin >> n;
	do {
		string a, b;
		cin >> a >> b;
		cout << apb(a, b) << endl;
	} while (--n);
}